src/cc/bcc_syms.cc

```diff
@@ -291,7 +299,7 @@ void ProcSyms::Module::load_sym_table() {
   if (type_ == ModuleType::PERF_MAP)
     bcc_perf_map_foreach_sym(name_.c_str(), _add_symbol, this);
   if (type_ == ModuleType::EXEC || type_ == ModuleType::SO)
-    bcc_elf_foreach_sym(name_.c_str(), _add_symbol, symbol_option_, this);
+    bcc_elf_foreach_sym_lazy(name_.c_str(), _add_symbol_lazy, symbol_option_, this);
```

Comment:
Do you meant to just change the default behavior? Would be great if it is configurable and default behavior kept the same?

Reply:
I originally had the extra option and kept the default the same. @yonghong-song suggested to change the default. I can do either -- just let me know.

Reply:
I suggest to change the default as it uses less memory and generally good. But if we are afraid of any breakage, we can keep the original as default, make the new implementation configurable, and after some baking and testing, and then flip.

Reply:
I wish we could keep the default behavior. There are a lot of possibilities of where an executable is (like it could be in NFS etc.) and going to the fs for every symbol may not be a very good default thing to do?